### PR TITLE
Remove old proposed accelerator interfaces

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
The driver interfaces described in `crypto_accel_driver.h` and `crypto_entropy_driver.h` are no longer being worked on. We do not intend to finish the design of these interfaces or to implement them in Mbed TLS. They have been superseded by the [unified driver interface](https://github.com/ARMmbed/mbedtls/blob/development/docs/proposed/psa-driver-interface.md), which is being actively worked on both to finalize the specification and to implement it in Mbed TLS.
    
The partially implemented dynamic secure element interface is staying for now.

### CODE REVIEW

File: docs/architecture/testing/driver-interface-test-strategy.md [hunk 1]
```diff
@@ -4,9 +4,19 @@ This document describes the test strategy for the driver interfaces in Mbed Cryp
...
+## Secure element driver interface testing
 
-The secure element driver interface (SE interface for short) is defined by [`psa/crypto_se_driver.h`](../../../include/psa/crypto_se_driver.h). This is an interface between Mbed Crypto and one or more third-party drivers.
+### Secure element driver interfaces
+
+#### Opaque driver interface
+
+The [unified driver interface](../../proposed/psa-driver-interface.md) supports both transparent drivers (for accelerators) and opaque drivers (for secure elements).
+
+Drivers exposing this interface needs to be registered at compile time by declaring their JSON description file.
```
Comment (ronald-cron-arm) [inline docs/architecture/testing/driver-interface-test-strategy.md:15]: s/needs/need

**Review by ronald-cron-arm — APPROVED** [review]
LGTM. I've checked that there is no mention anymore in the library to the symbols that were defined by `crypto_accel_driver.h` and `crypto_entropy_driver.h`. I've only found a typo, approving anyway.

File: docs/architecture/testing/driver-interface-test-strategy.md (hunk 1 above)
Comment (gilles-peskine-arm) [reply docs/architecture/testing/driver-interface-test-strategy.md]: Fixed by amending the corresponding commit.

Comment (ronald-cron-arm) [reply docs/architecture/testing/driver-interface-test-strategy.md]: Great.

**Review by mpg — APPROVED** [review]
LGTM

**Pull request merged**